File: network/pyjava_servers/TCPServer.py
#!/usr/bin/python
#TCPServer.py
from socket import *
serverPort = 7777
serverSocket = socket(AF_INET,SOCK_STREAM)
serverSocket.bind(("",serverPort))
serverSocket.listen(1)
print ("The TCP server is ready to receive ... ")

while 1:
    connectionSocket, addr = serverSocket.accept()
    sentence = connectionSocket.recv(1024).decode()
    print ("Received From Client: ", sentence)
    capitalizedSentence = sentence.upper()
    connectionSocket.send(capitalizedSentence.encode())
    print ("Sent back to Client: ", capitalizedSentence)
    connectionSocket.close()

File: network/pyjava_servers/TCPServer.java
import java.io.*;
import java.net.*;

class TCPServer{
    public static void main(String[] argv)throws Exception{
	String clientSentence,capitalizedSentence;
	ServerSocket welcomeSocket=new ServerSocket(7777);
	System.out.println("TCP Server Listening on port 7777");
	while(true){
	    Socket connectionSocket=welcomeSocket.accept();
	    BufferedReader inFromClient=new BufferedReader(new InputStreamReader(connectionSocket.getInputStream()));
	    DataOutputStream outToClient=new DataOutputStream(connectionSocket.getOutputStream());
	    clientSentence=inFromClient.readLine();
	    System.out.println("Received from client: "+ clientSentence );
	    capitalizedSentence=clientSentence.toUpperCase()+"\n";
	    outToClient.writeUTF(capitalizedSentence);
	    System.out.println("Sent back to client: "+ capitalizedSentence);
	}
    }
}
